packages/react-router/lib/rsc/browser.tsx

```diff
@@ -54,13 +54,12 @@ export type EncodeReplyFunction = (
   options: { temporaryReferences: unknown }
 ) => Promise<BodyInit>;
 
-declare global {
-  interface Window {
+type WindowWithRouterGlobals = Window &
+  typeof globalThis & {
     __router: DataRouter;
     __routerInitialized: boolean;
     __routerActionID: number;
-  }
-}
+  };
```

Comment:
The work in this PR triggered type errors, seemingly due to the global types leaking out of the library. As a workaround I introduced this to avoid touching global types.